File: SimpleDynamicEncryption.js
class SimpleDynamicEncryption {
    static Encode(strInput) {
        let rand = Math.floor(Math.random() * 9) + 1;

        let result = "";
        for (let i = 0; i < strInput.length; i++) {
            result += String.fromCharCode(strInput.charCodeAt(i) + rand + 2);
        }

        result += String.fromCharCode(rand + 33);

        return result;
    }

    static Decode(strInput) {
        let rand = strInput.charCodeAt(strInput.length - 1) - 33;

        strInput = strInput.substring(0, strInput.length - 1);

        let result = "";
        for (let i = 0; i < strInput.length; i++) {
            result += String.fromCharCode(strInput.charCodeAt(i) - rand - 2);
        }

        return result;
    }
}

// Example usage:
var input_string = "Mehdi-RaTo";
var encoded_string = SimpleDynamicEncryption.Encode(input_string);
var decoded_string = SimpleDynamicEncryption.Decode(encoded_string);

console.log("Original: ", input_string);
console.log("Encoded:  ", encoded_string);
console.log("Decoded:  ", decoded_string);

File: SimpleDynamicEncryption.py
import random

class SimpleDynamicEncryption:
    @staticmethod
    def Encode(strInput):
        rand = random.randint(1, 9)

        result = ""
        for char in strInput:
            result += chr(ord(char) + rand + 2)

        result += chr(rand + 33)

        return result

    @staticmethod
    def Decode(strInput):
        rand = ord(strInput[-1]) - 33

        strInput = strInput[:-1]

        result = ""
        for char in strInput:
            result += chr(ord(char) - rand - 2)

        return result

# Example usage:
input_string = "Mehdi-RaTo"
encoded_string = SimpleDynamicEncryption.Encode(input_string)
decoded_string = SimpleDynamicEncryption.Decode(encoded_string)

print("Original: ", input_string)
print("Encoded:  ", encoded_string)
print("Decoded:  ", decoded_string)
